File: Torneig_Mort_Spring/src/main/java/org/eljust/DTO/EquipGrupDTO.java
package org.eljust.DTO;

import lombok.Getter;
import lombok.Setter;

/**
 * Aquest DTO d'Equip l'hem creat per a poder mostrar els equips d'un grup i no
 * caure amb la recursivitat infinita de grup i equips, amés tampoc mostrarà els
 * jugadors dels equpis. Simplifiquem la busqueda. El farem servir al GrupDTO i
 * GrupEquipDTO
 */
@Setter
@Getter
public class EquipGrupDTO {
	private Long idEquip;
	private String nom;
	private String curs;
	private String imatge;
	private boolean esGuanyador;
	private int punts;
	private int puntsContra;
	private int partitsJugats;
	private int partitsGuanyats;
	private int partitsPerduts;
	private int partitsEmpatats;

}

File: app_torneig_flutter/lib/models/equip_grup_dto.dart
class EquipGrupDTO {
  int? idEquip;
  String? nom;
  String? curs;
  String? imatge;
  bool? esGuanyador;
  int? puntsFavor;
  int? puntsContra;
  int? partitsJugats;
  int? partitsGuanyats;
  int? partitsEmpatats;
  int? partitsPerduts;

  EquipGrupDTO({
    required this.idEquip,
    this.nom,
    this.curs,
    this.imatge,
    this.esGuanyador,
    this.puntsFavor,
    this.puntsContra,
    this.partitsJugats,
    this.partitsGuanyats,
    this.partitsEmpatats,
    this.partitsPerduts,
  });

  EquipGrupDTO.fromJSON(Map<String, dynamic> objecteJSON) {
    idEquip = objecteJSON['idEquip'];
    nom = objecteJSON['nom'] ?? "";
    curs = objecteJSON['curs'] ?? "";
    imatge = objecteJSON['imatge'] ?? "assets/images/defecte.png";
    esGuanyador = objecteJSON['esGuanyador'];
    puntsFavor = objecteJSON['punts'];
    puntsContra = objecteJSON['puntsContra'];
    partitsJugats = objecteJSON['partitsJugats'];
    partitsGuanyats = objecteJSON['partitsGuanyats'];
    partitsEmpatats = objecteJSON['partitsEmpatats'];
    partitsPerduts = objecteJSON['partitsPerduts'];
  }
}
